### DISCUSSION TITLE
PNG to PWMO/Photon?

### DISCUSSION THREAD

**Discussion by takigama** [sn4k3/UVtools#165] [Q&A] [answered]
UVTools has the ability to load images (png's, etc), but I cant quite see what its for, but it would be nice to be able to convert an image to a PWMO file (or other printer format) but if there is a way to do it I cant seem to find it.

I used to use a photon to print PCB's, and part of that process was using "PNGtoPhoton" which would just produce a one layer print that had the PCB as an image, I did attempt this, but for some reason UVtools seem to not be able to convert the resulting photon to a PWMO file.

UVtools seems to load the photon file ok, but when I convert it to a PWMO file it looks ok in UVtools but its definitely wrong on the printer, its seemed to tile the image across one axis of the printer

Any suggestions?

From uvtools:
![Screenshot from 2021-03-13 03-17-35](https://user-images.githubusercontent.com/151283/110968017-2054b280-83ab-11eb-8cac-767e2bd1885c.png)

On the mono:
![20210313_025700_crop](https://user-images.githubusercontent.com/151283/110968336-80e3ef80-83ab-11eb-8c58-337c6da8c57f.jpg)

**Comment by sn4k3** [comment, marked as answer]
Load png's/images are just to do some processing on the image, they can't be used to convert into formats.
However you can do what you want via other via.

1. Load a previous good sliced file for your printer on UVtools (.PWMO)
2. Go to Actions -> Import layers
3. Import type: Insert
4. Select image(s) to import (if more than one, the resolution must be the same)
5. Discard unmodified layers
6. Hit apply

Example:
![image](https://user-images.githubusercontent.com/113281/110973453-2f107a80-8355-11eb-9f93-caf681274dc3.png)

Result:
![image](https://user-images.githubusercontent.com/113281/110974149-1e143900-8356-11eb-8e8f-256bfbbfe752.png)

**Reply by takigama** [reply]
Thanks, that did indeed work much better!

![20210314_011534_crop](https://user-images.githubusercontent.com/151283/111033091-a5030780-8463-11eb-81ed-8bd9c7bb8645.jpg)

**Reply by takigama** [reply]
Had an odd problem the second time I tried this.

Initially used an original pwmo file from photon workshop to generate the one above (removed all the layers and just inserted my new layer, so the whole print only had one layer).

Exported that to a file called "base.pwmo" - worked fine on the printer.

I then created another png, replaced layer one with uvtools, saved the pwmo file, but the resulting pwmo file was identical to base.pwmo (i.e. nothing changed).

I then did the same above but clnoed the layer a couple of times and this time it worked. Not sure why UVtools fails when I replaced a single layer but it was a bit odd.

**Reply by sn4k3** [reply]
You saved the file after do the operations right? (File - Save)?
I may have an ideia, let me check...

EDIT: Found the problem and will be fixed next patch.
To temporary get arround just use pixel editor, remove a pixel from black background, hit apply and save the file

**Reply by takigama** [reply]
Thanks again, For a while I thought I was going insane! :+1:
